src/renderer/components/AgentBase.vue

```diff
@@ -103,14 +103,16 @@ import message_bus from '@/message_bus.js';
 import share, {share_source} from '@/share.js';
 import components, {shared} from './components.js';
 
-// icons from https://element.eleme.io/#/en-US/component/icon
+// The (. && .. && ...) || 'default' syntax provides defaults for modules that lack any level of the metadata
+//  definition. It would be useful if javascript had an Elvis Operator, but it does not.
+// The icons from https://element.eleme.io/#/en-US/component/icon
 let module_list = Object.entries(components).map(([modulename, module]) => ({
   path: module.default.name,
-  label: module.metadata && module.metadata.menu && module.metadata.menu.label ? module.metadata.menu.label : module.default.name,
-  icon: module.metadata && module.metadata.menu && module.metadata.menu.icon ? module.metadata.menu.icon : 'el-icon-document',
-  group: module.metadata && module.metadata.menu && module.metadata.menu.group ? module.metadata.menu.group : 'Other',
-  priority: module.metadata && module.metadata.menu && module.metadata.menu.priority ? module.metadata.menu.priority : 100,
-  required_protocols: module.metadata && module.metadata.menu && module.metadata.menu.required_protocols ? module.metadata.menu.required_protocols : [],
+  label: (module.metadata && module.metadata.menu && module.metadata.menu.label) || module.default.name,
```

Comment:
Small change but this actually made it much easier for me to understand at least lol. Thanks.